### PR TITLE
Controllers spec

### PR DESCRIPTION
**Pull request by aradradev**
**Pull Request: In this project, I've implemented the following features:**

- Add RSpec as a gem to the blog-app project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
- If the response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

Please review this pull request and give your feedback if necessary.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,7 +1,31 @@
 require 'rails_helper'
 
-RSpec.describe 'Posts', type: :request do
+RSpec.describe PostsController, type: :controller do
```
Comment (descholar-ceo) [inline spec/requests/posts_spec.rb]: - [ ] You were supposed to make type as `:request` not ~:controller~

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -1,7 +1,29 @@
 require 'rails_helper'
 
-RSpec.describe 'Users', type: :request do
+RSpec.describe UsersController, type: :controller do
```
Comment (descholar-ceo) [inline spec/requests/users_spec.rb]: - [ ] You were supposed to make type as `:request` not ~:controller~

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
+      user = User.create(name: 'Jalloh', posts_counter: 0)
+      get :show, params: { id: user.id }
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'it renders the show template' do
+      user = User.create(name: 'John', posts_counter: 0)
+      get :show, params: { id: user.id }
+      expect(response).to render_template(:show)
+    end
```
Comment (descholar-ceo) [inline spec/requests/users_spec.rb]: - [ ] You are supposed to include tests that ensures the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -1,7 +1,31 @@
...
+      get :show, params: { id: post.id }
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'returns the correct template show' do
+      user = User.create(name: 'Doe')
+      post = Post.create(author: user, title: 'Hello', text: 'Sample text')
+      get :show, params: { id: post.id }
+      expect(response).to render_template(:show)
+    end
```
Comment (descholar-ceo) [inline spec/requests/posts_spec.rb]: - [ ] You are supposed to include tests that ensures the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @aradradev 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
👍🏽 The setup is done correctly ✔️ 
👍🏽 The test you wrote are good and they work, except few improvements that you need to add ✔️ 
👍🏽 The PR follows the correct git flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Installing rspec tests to run through github actions so that you see your tests running at every push ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @aradradev 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 4 above)
Comment (aradradev) [reply spec/requests/posts_spec.rb]: @descholar-ceo Thanks for the tips, always good reviewers in this module.

**Pull request merged**